lnwallet/btcwallet/signer.go

```diff
@@ -403,7 +403,13 @@ func (b *BtcWallet) SignOutputRaw(tx *wire.MsgTx,
 			}
 		}
 
-		sig, err := schnorr.ParseSignature(rawSig)
+		// The signature returned above might have a sighash flag
+		// attached if a non-default type was used. We'll slice this
+		// off if it exists to ensure we can properly parse the raw
+		// signature.
+		sig, err := schnorr.ParseSignature(
+			rawSig[:schnorr.SignatureSize],
```

Comment:
Should we check the length first to make sure we don't run into an out of bounds panic?

Reply:
The `rawSig` returned from `txscript.RawTxInTapscriptSignature` (10 lines above) is documented to be a "valid schnorr signature", so perhaps it's safe to assume that if no `err` was returned then the length check isn't necessary

Reply:
If `txscript.RawTxInTapscriptSignature` returns without an `err` would it be safe to assume there wouldn't be any length issues?

It's not explicitly mentioned in the godoc but the function seems to only return valid schnorr signatures.

Reply:
Yes, but because there's no `default` case in the switch over `signDesc.SignMethod`, we could end up not calling any signing function. Probably that's not a reachable path with the current signing methods and earlier checks, but I think better safe than `panic()`...

Reply:
Modified things to add a default case, so we'll error out now due to an unknown signing method. 

The two current function calls will always return an sig of at least 64-byte  bytes. Otherwise, `schnorr.ParseSignature` itself wouldn't have worked. 

I went to write a test then found a bug, which is fixed here: https://github.com/btcsuite/btcd/pull/1941